helm/cluster/templates/apps/apps.yaml

```diff
@@ -70,7 +70,12 @@ spec:
       name: {{ $.Values.global.metadata.name }}-kubeconfig
       namespace: {{ $.Release.Namespace}}
   {{- end }}
-  version: {{ $ephemeralConfig.versionOverride | default $app.version | trim | trimPrefix "v" }}
+  {{- $appVersion := $app.version -}}
+  {{- if $.Values.providerIntegration.useReleases }}
+  {{- $_ := set $ "appName" $app.appName }}
+  {{ $appVersion = include "cluster.app.version" $ }}
+  {{- end }}
+  version: {{ $ephemeralConfig.versionOverride | default $appVersion | trim | trimPrefix "v" }}
```

Comment:
Unrelated to Release CR, it is still possible to override app version like before (e.g. for the purposes of development, putting out on-call fire and similar).